packages/legend-query/src/stores/filterOperators/QueryBuilderFilterOperatorHelper.ts

```diff
@@ -328,56 +363,100 @@ const buildFilterConditionStateWithExists = (
       filterState.queryBuilderState.graphManagerState.graph.getTypicalMultiplicity(
         TYPICAL_MULTIPLICITY_TYPE.ONE,
       );
+    let flattenedPropertyExpressionChain: AbstractPropertyExpression;
+
     const initialPropertyExpression = guaranteeType(
-      expression.parametersValues[0],
+      precedingExpression.parametersValues[0],
       AbstractPropertyExpression,
     );
-    let propertyExpression = new AbstractPropertyExpression(
+    const propertyExpression = new AbstractPropertyExpression(
```

Comment:
this is a silly comment, but we would prefer to do

```ts
flattenedPropertyExpressionChain = new AbstractPropertyExpression(..
```

In other words, not creating this variable that we don't need. The reason being this variable belongs to the outer scope, later on in the deeper scope, you create another variable of the same name `propertyExpression`. Let's avoid doing this to avoid confusion and potential bugs (this is a Sonar thing, but it grows on me more and more). Thanks!